### ISSUE TITLE
how to alleviate reward hacking?

### ISSUE THREAD

**Issue by MengHao666** [yifan123/flow_grpo#233]
I use the flow-grpo algorithm on a new model, with the code modified with [scripts/train_qwenimage.py](https://github.com/yifan123/flow_grpo/blob/main/scripts/train_qwenimage.py).

but in the only few epochs, model tend to reward hacking, with very small grids on human face and background.

I am trying to increase the weight of kl-loss on velocity, as recommended from the author.
Are there any effective tircks to alleviate such kind of reward hacking problem?

<img width="1792" height="2304" alt="Image" src="https://github.com/user-attachments/assets/621c3c7a-91ca-40e5-ae52-ef295672a9fc" />

<img width="1404" height="1864" alt="Image" src="https://github.com/user-attachments/assets/9cebf7a0-4a14-4c4e-bb72-59e224cb46fa" />

![Image](https://github.com/user-attachments/assets/38e3be81-48c6-43c5-a542-b76cc96b8108)

![Image](https://github.com/user-attachments/assets/e3dbc5bb-567d-4c49-a742-872d68beb777)

![Image](https://github.com/user-attachments/assets/ca6ae408-10a6-4e65-b756-d3572180eeea)

**Comment by MengHao666** [comment]
the main train logs are like the following:

the 3 curves only differ in learning rate.

<img width="3094" height="1630" alt="Image" src="https://github.com/user-attachments/assets/94cee934-d972-436a-b286-27bf429d083e" />

**Comment by lilyandluc** [comment]
i can not understand your explanation on reward hacking, what does this lie in ? the human looks very vivid
